protocol/messenger_mailserver_cycle.go

```diff
@@ -216,33 +189,85 @@ func (m *Messenger) allMailservers() ([]mailservers.Mailserver, error) {
 
 		for _, c := range customMailservers {
 			if c.Fleet == fleet {
-				c.Version = m.transport.WakuVersion()
 				allMailservers = append(allMailservers, c)
 			}
 		}
 	}
 
-	// Filter mailservers by configured waku version
-	wakuVersion := m.transport.WakuVersion()
-	matchingMailservers := make([]mailservers.Mailserver, 0, len(allMailservers))
+	return allMailservers, nil
+}
 
-	for _, ms := range allMailservers {
-		if ms.Version == wakuVersion {
-			matchingMailservers = append(matchingMailservers, ms)
+type SortedMailserver struct {
+	Mailserver      mailservers.Mailserver
+	RTT             time.Duration
+	CanConnectAfter time.Time
+}
+
+func (m *Messenger) getAvailableMailserversSortedByRTT(allMailservers []mailservers.Mailserver) []mailservers.Mailserver {
+	// TODO: this can be replaced by peer selector once code is moved to go-waku api
+	availableMailservers := make(map[string]time.Duration)
+	availableMailserversMutex := sync.Mutex{}
+	availableMailserversWg := sync.WaitGroup{}
+	for _, mailserver := range allMailservers {
+		availableMailserversWg.Add(1)
+		go func(mailserver mailservers.Mailserver) {
+			defer availableMailserversWg.Done()
+
+			peerID, err := mailserver.PeerID()
+			if err != nil {
+				return
+			}
+
+			ctx, cancel := context.WithTimeout(m.ctx, 500*time.Millisecond)
+			defer cancel()
+
+			rtt, err := m.transport.PingPeer(ctx, peerID)
```

Comment:
This was created temporarilly to remove the ping code from mailservers.
This will be replaced by fastest peer selection once this is moved to go-waku